### PR TITLE
Lagrange polynomial

### PR DESCRIPTION
**Pull request by JEAlfonsoP**
Hi, please review whenever you can and let me know as much of your feedback , since this is the first attempt to PR. I tried to follow as much as possible existing sections style. Did not want to repeat LP equation all the way down in the written.

### CODE REVIEW

**Comment by JEAlfonsoP** [comment]
> Hi, please review whenever you can and let me know as much of your feedback , since this is the first attempt to PR. I tried to follow as much as possible existing sections style. Did not want to repeat LP equation all the way down in the written. 
> 
> 

@kevaundray ^

File: docs/Theory/Math/lagrange polynomial.md [hunk 1]
```diff
@@ -1 +1,51 @@
...
+
+$$\sum_{j=0}^{k} {y_j*l_j(x)}$$
+
+Where $l_j(x)$ is Langrange basis (polynomial): 
+ 
+$$
+\mathcal{l_j}(x) = \prod_{m \neq j, m >= 0}\frac{x -x_m}{x_j - x_m}
+$$
+
+As mentioned above the interpolating polynomial is unique. This characteristic allows uses in cryptography such as: Shamir's secret sharing scheme and KZC polynomial commitmens (Kate commitment) *[https://en.m.wikipedia.org/wiki/Lagrange_polynomial]*
```
Comment (kevaundray) [inline docs/Theory/Math/lagrange polynomial.md:18]: The commitment scheme is referred to as KZG.

Typo: commitmens -> commitments

Can you add a reference for KZG?

File: docs/Theory/Math/lagrange polynomial.md [hunk 2]
```diff
@@ -1 +1,51 @@
...
+
+## Uses
+
+### Vector commitment and Verkle Tries
+
+A commitment scheme allows one to commit to a chosen value (or statement) while keeping it hidden to others, with the ability to reveal the committed value later *[https://en.m.wikipedia.org/wiki/Commitment_scheme]*.
+
+In Verkle Tree, the commitment method is called vector commitment *[https://hackmd.io/@sin7y/rJZZy_mD9]*.
```
Comment (kevaundray) [inline docs/Theory/Math/lagrange polynomial.md:26]: This is not the correct reference for vector commitments. You can use the original verkle tree paper

File: docs/Theory/Math/lagrange polynomial.md (hunk 1 above)
Comment (JEAlfonsoP) [reply docs/Theory/Math/lagrange polynomial.md]: I added the original reference for KZG:

https://www.iacr.org/archive/asiacrypt2010/6477178/6477178.pdf

File: docs/Theory/Math/lagrange polynomial.md (hunk 2 above)
Comment (JEAlfonsoP) [reply docs/Theory/Math/lagrange polynomial.md]: I added the original paper for Verkel Trees:

https://math.mit.edu/research/highschool/primes/materials/2018/Kuszmaul.pdf (John Kuszmaul).

Let me know if you would like one more reference or a different.

**Pull request merged**